### PR TITLE
Refactoring testing

### PR DESCRIPTION
**Pull request by FlorianKutz**
**What issue type does this pull request address?** (keep at least one, remove the others)  
/kind enhancement

**What is this pull request for? Which issues does it resolve?** (use `resolves #<issue_number>` if possible)  
This pull requests adds:
- more refactoring
- more tests
- resolves a bug that made it impossible to remove images from the config completely.

### CODE REVIEW

File: pkg/devspace/config/loader/save.go [hunk 1]
```diff
@@ -73,6 +73,17 @@ func (l *configLoader) RestoreVars(config *latest.Config) (*latest.Config, error
 	return clonedConfig, nil
 }
 
+var keysNoRestoration = []string{"images", "deployments", "dev"}
```
Comment (FabianKramm) [inline pkg/devspace/config/loader/save.go]: rename to ignoreConfigKeys

File: pkg/devspace/build/create_builder.go [hunk 2]
```diff
@@ -16,13 +16,19 @@ import (
 	"github.com/pkg/errors"
 )
 
+var fakeBuilder builder.Interface
```
[the remark this answers is not in the document]
Comment (FlorianKutz) [reply pkg/devspace/build/create_builder.go]: Fixed with commit 1adc5ebc5c5897c2fe814dd35a918c106e7d238f

File: pkg/devspace/cloud/cluster.go [hunk 3]
```diff
@@ -80,7 +80,9 @@ func (p *provider) ConnectCluster(options *ConnectClusterOptions) error {
 	}
 
 	// Check what kube context to use
-	if options.KubeContext == "" {
```
[the remark this answers is not in the document]
Comment (FlorianKutz) [reply pkg/devspace/cloud/cluster.go]: Fixed in commit cd8e0b7b5d714e54fd6a344a14a6de3364aa892e . This also removes unnexxessary kubeClient parameters from private functions.

File: pkg/devspace/config/loader/save.go [hunk 4]
```diff
@@ -73,6 +73,17 @@ func (l *configLoader) RestoreVars(config *latest.Config) (*latest.Config, error
...
+
+func (l *configLoader) shouldRestoreKey(key interface{}) bool {
```
[the remark this answers is not in the document]
Comment (FlorianKutz) [reply pkg/devspace/config/loader/save.go]: Fixed in d2bb385c4962df17ae98edfac5800d7ee971952a .

File: pkg/devspace/config/loader/save.go (hunk 1 above)
Comment (FlorianKutz) [reply pkg/devspace/config/loader/save.go]: Fixed in ae4cc7473c71012c13d04ff7e0852409b14d34af

**Pull request merged**